tools/calyx-writer/lib.rs

```diff
@@ -223,32 +228,31 @@ pub struct Port {
     /// signature.
     parent: Option<String>,
     name: String,
-    /// A width of `-1` means the actual width is unknown/irrelevant/inferred.
-    width: isize,
+    width: Option<NonZeroU64>,
 }
 
 impl Port {
     /// Constructs a new port with the given `parent` and `name`.
     ///
-    /// Requires: `width > 0`.
+    /// Requires: `width` is nonzero.
     pub fn new<S: ToString, T: ToString>(
         parent: Option<S>,
         name: T,
-        width: usize,
+        width: u64,
     ) -> Self {
-        assert!(width > 0);
+        assert!(width != 0);
         Self {
             attributes: vec![],
             parent: parent.map(|s| s.to_string()),
             name: name.to_string(),
-            width: width as isize,
+            width: Some(unsafe { NonZeroU64::new_unchecked(width) }),
```

Comment:
Just use `NonZeroU64::new(width).expect("width cannot be zero")` and remove the assert. As a general rule, you should avoid using `unsafe` unless it is absolutely necessary and I don't think it is here.